packages/medusa/src/services/invite.ts

```diff
@@ -0,0 +1,277 @@
+import { BaseService } from "medusa-interfaces"
+import jwt, { JwtPayload } from "jsonwebtoken"
+import config from "../config"
+import { MedusaError } from "medusa-core-utils"
+import { User } from ".."
+import { EntityManager } from "typeorm"
+import { EventBusService, UserService } from "."
+import { InviteRepository } from "../repositories/invite"
+import { UserRepository } from "../repositories/user"
+import { ListInvite } from "../types/invites"
+import { UserRoles } from "../models/user"
+
+type InviteServiceProps = {
+  manager: EntityManager
+  userService: UserService
+  userRepository: UserRepository
+  inviteRepository: InviteRepository
+  eventBusService: EventBusService
+}
+
+class InviteService extends BaseService {
+  static Events = {
+    CREATED: "invite.created",
+  }
+
+  private manager_: EntityManager
+  private userService_: UserService
+  private userRepo_: UserRepository
+  private inviteRepository_: InviteRepository
+  private eventBus_: EventBusService
+
+  constructor({
+    manager,
+    userService,
+    userRepository,
+    inviteRepository,
+    eventBusService,
+  }: InviteServiceProps) {
+    super()
+
+    /** @private @constant {EntityManager} */
+    this.manager_ = manager
+
+    /** @private @constant {UserService} */
+    this.userService_ = userService
+
+    /** @private @constant {UserRepository} */
+    this.userRepo_ = userRepository
+
+    /** @private @constant {InviteRepository} */
+    this.inviteRepository_ = inviteRepository
+
+    /** @private @const {EventBus} */
+    this.eventBus_ = eventBusService
+  }
+
+  withTransaction(manager): InviteService {
+    if (!manager) {
+      return this
+    }
+
+    const cloned = new InviteService({
+      manager,
+      inviteRepository: this.inviteRepository_,
+      userService: this.userService_,
+      userRepository: this.userRepo_,
+      eventBusService: this.eventBus_,
+    })
+
+    cloned.transactionManager_ = manager
+
+    return cloned
+  }
+
+  async list(selector, config = {}): Promise<ListInvite[]> {
+    const inviteRepo = this.manager_.getCustomRepository(InviteRepository)
+
+    const query = this.buildQuery_(selector, config)
+
+    const invites = await inviteRepo.find(query)
+
+    return invites.map((inv) => {
+      return {
+        token: this.generateToken({
+          invite_id: inv.id,
+          role: inv.role,
+          user_email: inv.user_email,
+        }),
+        ...inv,
+      }
+    })
```

Comment:
Thinking we should only generate the token once; namely when we create the invite. Otherwise it will never expire?

Reply:
I have resolved this as we discussed